scripts/generate_syncset.py

```diff
@@ -27,33 +55,72 @@ def get_all_yaml_obj(file_paths):
+    o = copy.deepcopy(obj)
+    # Get all yaml files as array of yaml objects
+    yamls = get_all_yaml_obj(get_all_yaml_files(directory))
+    if len(yamls) == 0:
+        return
+
+    yamls = sorted(yamls)
+    for y in yamls:
+        if 'patch' in y:
+            if not 'patches' in o['spec']:
+                o['spec']['patches'] = []
+            o['spec']['patches'].append(y)
+        else:
+            if not 'resources' in o['spec']:
+                o['spec']['resources'] = []
+            o['spec']['resources'].append(y)
+
+    o['metadata']['name'] = name
+    # append object to the template's objects
+    template_data['objects'].append(o)
+
 
 if __name__ == '__main__':
     #Argument parser
     parser = argparse.ArgumentParser(description="selectorsyncset generation tool", usage='%(prog)s [options]')
-    parser.add_argument("--template-path", "-t", required=True, help="Path to template file [required]")
+    parser.add_argument("--template-dir", "-t", required=True, help="Path to template directory [required]")
     parser.add_argument("--yaml-directory", "-y", required=True, help="Path to folder containing yaml files [required]")
     parser.add_argument("--destination", "-d", required=True, help="Destination for selectorsynceset file [required]")
-    parser.add_argument("--git-hash", "-c", required=True, help="Commit hash of commit used [required]")
+    parser.add_argument("--repo-name", "-r", required=True, help="Name of the repository [required]")
     arguments = parser.parse_args()
 
-    # Get all yaml files as array of yaml objecys
-    yamls = get_all_yaml_obj(get_all_yaml_files(arguments.yaml_directory))
     # Get the template data
-    template_data = get_yaml(arguments.template_path)
-    # Configure template
-    for obj in template_data['objects']:
-        if not 'labels' in obj['metadata']:
-            obj['metadata']['labels'] = {}
-        # create labels
-        obj['metadata']['labels']['managed.openshift.io/osd'] = "true"
-        obj['metadata']['labels']['managed.openshift.io/gitHash'] = arguments.git_hash
-
-        # create resources
-        obj['spec']['resources'] = yamls
-    
-    # write selectorsyncset file
+    template_data = get_yaml(os.path.join(arguments.template_dir, "template.yaml"))
+    selectorsyncset_data = get_yaml(os.path.join(arguments.template_dir, "selectorsyncset.yaml"))
+
+    # The templates and script are shared across repos (copy & paste).
+    # Set the REPO_NAME parameter.
+    for p in template_data['parameters']:
+        if p['name'] == 'REPO_NAME':
+            p['value'] = arguments.repo_name
+
+    # for each subdir of yaml_directory append 'object' to template
+    for (dirpath, dirnames, filenames) in os.walk(arguments.yaml_directory):
+        if filenames:
+            sss_name = dirpath.replace('/','-')
+            if sss_name == arguments.yaml_directory:
+                # files in the root dir, use repo-name for SSS name
```

Comment:
nit: if variable is abbreviated, might be helpful to just spell out in comments. I had a moment of "what is SSS"